### PR TITLE
Fix value provider assumption

### PR DESCRIPTION
**Pull request by activatedgeek**
- Treats all inputs as strings instead of `ValueProviders`
- Support extraneous data in `PredictionDoFn`

/cc @yixinshi @jlewi

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/batch-predict/9)

### CODE REVIEW

**Comment by activatedgeek** [comment]
/hold

Instead of passing full dicts, SideInputs is the right way I think. Making that change.

**Comment by activatedgeek** [comment]
/hold cancel

Side Inputs are even more restrictive. So not doing that.

File: kubeflow_batch_predict/dataflow/batch_prediction.py [hunk 1]
```diff
@@ -207,7 +211,9 @@ def process(self, element, model_dir):
           loaded_data = [json.loads(d) for d in element]
       else:
         loaded_data = [json.loads(d) for d in element]
-      instances = mlprediction.decode_base64(loaded_data)
+      loaded_data = mlprediction.decode_base64(loaded_data)
+      instances = loaded_data[self._instances_key]
```
Comment (yixinshi) [inline kubeflow_batch_predict/dataflow/batch_prediction.py]: To make it compatible to the case that there is no "_instances_key", put a if here or an abstract class here. Also, try to "rowfy" the inputs by transforming it to list of 
[{"instance_key": value1, "meta1": xx, "meta2": yy}, {"instance_key": value2, "meta1": xx, "meta2": yy}]. This way, we can do multi-fanout in line 235.

File: kubeflow_batch_predict/dataflow/batch_prediction.py [hunk 2]
```diff
@@ -226,8 +232,8 @@ def process(self, element, model_dir):
           td.microseconds / 10**3  + (td.seconds + td.days * 24 * 3600) * 10**3)
       self._batch_process_ms_distribution.update(time_delta_in_ms)
 
-      for i, p in zip(inputs, predictions):
-        yield i, p
+      loaded_data[self._predictions_key] = predictions
```
Comment (yixinshi) [inline kubeflow_batch_predict/dataflow/batch_prediction.py]: If you do the suggestion in line 214, you need to do a "zip" to match the predictions with other "passthrough" fields.

**Comment by jlewi** [comment]
Stale

**Pull request closed without merging**